pkg/kubectl/cmd/set/set_selector.go

```diff
@@ -213,6 +213,9 @@ func updateSelectorForObject(obj runtime.Object, selector metav1.LabelSelector)
 
 // getResourcesAndSelector retrieves resources and the selector expression from the given args (assuming selectors the last arg)
 func getResourcesAndSelector(args []string) (resources []string, selector *metav1.LabelSelector, err error) {
+	if len(args) == 0 {
```

Comment:
We don't need this in Complete. The check that was added by @xilabao in Validate (`o.selector == nil`) should be enough.